### DISCUSSION TITLE
PR broke my distribution pipeline

### DISCUSSION THREAD

**Discussion by denehoffman** [PyO3/maturin#2146] [Q&A] [answered]
[PR 2125](https://github.com/PyO3/maturin/pull/2125) "fixes" issues with duplicate files in Python wheels. Except it also seems to break sdist. I have a workspace project organized like:

```
project/
project/README.md
project/crates/Cargo.toml (uses it's own README.md)
project/Cargo.toml (has workspace readme pointing to "README.md")
project/py-project/
project/py-project/pyproject.toml (points readme to "README.md")
project/py-project/README.md
project/py-project/Cargo.toml (points readme to "README.md")
```

I've tried just about every combination I can think of, but I keep getting
```
File project-x.y.z/README.md was already added from project/README.md, can't added it from project/py-project/README.md
```

despite the hastily spell-checked error message, I've tried every combination I can think of for README paths. The issue is that the only one that works is where I remove the `readme` line from my `pyproject.toml` entirely, which I believe will cause problems later on when I try to upload to PyPi. However, I'm not sure if this is the case, but either way, it should be documented. My build worked fine prior to this PR, and the other distribution methods (building wheels for different platforms) *still* works fine. Am I doing something wrong here?

Edit:
The documentation states that:

> maturin supports [PEP 621](https://www.python.org/dev/peps/pep-0621/), you can specify python package metadata in pyproject.toml. maturin merges metadata from Cargo.toml and pyproject.toml, pyproject.toml takes precedence over Cargo.toml.

If this is the case, I would expect the following behavior: Remove readme field from project/py-project/Cargo.toml, include it in pyproject.toml, everything works okay. However, I *still* get the error that the readme was added from the root directory, even though I mention it exactly nowhere in the py-project directory. This is clearly not the expected behavior as far as I'm concerned.

**Comment by konstin** [comment, marked as answer]
Could you share a minimal reproducible example? I tried reproducing your description but did not get your error: https://github.com/konstin/pyo3-2146

**Reply by denehoffman** [reply]
The repo in your reply does look like mine, so I'm not sure what I was doing wrong. I was able to resolve it by only mentioning the README in the Cargo.toml, letting maturin fill in the rest. Thank you for looking into it though, I appreciate it!
